systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java

```diff
@@ -243,13 +243,16 @@ public static void waitUntilPodsStability(List<Pod> pods) {
 
         TestUtils.waitFor("Pods stability", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
             () -> {
-                for (Pod pod : pods) {
+                List<Pod> actualPods = pods.stream().map(p -> kubeClient().getPod(p.getMetadata().getName())).collect(Collectors.toList());
+
+                for (Pod pod : actualPods) {
                     if (pod.getStatus().getPhase().equals("Running")) {
                         LOGGER.info("Pod {} is in the {} state. Remaining seconds pod to be stable {}",
                             pod.getMetadata().getName(), pod.getStatus().getPhase(),
                             Constants.GLOBAL_RECONCILIATION_COUNT - stabilityCounter[0]);
                     } else {
                         LOGGER.info("Pod {} is not stable with phase {}", pod.getMetadata().getName(), pod.getStatus().getPhase());
+                        stabilityCounter[0] = 0;
```

Comment:
Just a minor question. Why are you using an `int[]` here instead of just an `int`. You only ever refer to the first cell of the array and don't use the array outside of the lambda given to `waitFor`. Not saying you should change it, just interested why?

Reply:
He needs a final to use it within the lambda, but he also needs it to be mutable.

Reply:
Getting the actual pods is good, but don't we need to check that exactly the given `pods` are present in the `actualPods` too?

Reply:
The array with one cell is there because of one simple reason. If you look at the whole method, you will see that this variable needs to be changed whenever the BooleanSupplier condition is `TRUE`. In this case, with each tick(poll) the whole code will be executed. If pods are all in running state we will increment the stability counter. Moreover, you need some outside reference from the scope of Lambda. Use have two options:
- use atomic variables - which is not good approach to do it and should be used only in concurrent stuff...
- use array - which completely resolve that problem

Reply:
Well, see that I did not refresh and you were quicker @tombentley  :D

Reply:
I think I am missing something, this lambda just checks that all the pods are ready by counting how many are ready and comparing it to a constant, and repeating that at the configured interval, why does the counter need to be outside the lambda?

Reply:
If you use counter `inside` the lambda you will override him at each poll interval(tick).

Reply:
Yeah I get that, but isn't it set to zero every time a pod isn't ready anyway? Maybe this is something to do with the phases? 

Anyway, if you guys are happy, I'm happy.

Reply:
@tombentley I discuss that with @see-quick and it's enough from our POV. Now we can use it in situations when rolling update is not completely finished. We can override it for usage you suggested for tests, where we are for 100% sure that we don't want any rolling update during this wait.